### ISSUE TITLE
Dose anynone reproduce ShuffleNet in keras?

### ISSUE THREAD

**Issue by hellojialee** [keras-team/keras#7561]
An efficient CNN architecture: 
https://arxiv.org/pdf/1707.01083.pdf
How to realize the shuffle layer?

**Comment by jaxony** [comment]
I made a PyTorch version here: https://github.com/jaxony/ShuffleNet/blob/master/model.py
Have a look at the `channel_shuffle(x, groups)` function in the link. Let me know if you have more questions about how to translate it to Keras.

**Comment by BelalC** [comment]
@USTClj - did you manage to implement ShuffleNet in Keras? I'd be really interested to see your progress/contribute (or try out the model if you have already completed it!)

**Comment by Abhishek500** [comment]
Hi any updates here?
